version: "3.8"
services:
  db:
    image: postgres
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U user -d mydatabase"]
      interval: 10s
      timeout: 5s
      retries: 5
    volumes:
      - postgres_data:/var/lib/postgresql/data
    environment:
      POSTGRES_DB: mydatabase
      POSTGRES_USER: user
      POSTGRES_PASSWORD: password
    ports:
      - "5432:5432"

  pythonapp:
    build: .
    depends_on:
      db:
        condition: service_healthy
    environment:
      DATABASE_HOST: db
      DATABASE_NAME: mydatabase
      DATABASE_USER: user
      DATABASE_PASSWORD: password
    command: >
      sh -c "python app.py && python fetch_detailed_metadata.py"

volumes:
  postgres_data: